File: src/main/java/dk/cngroup/fetch/controller/ProductController.java
package dk.cngroup.fetch.controller;

import dk.cngroup.fetch.entity.Product;
import dk.cngroup.fetch.repository.ProductRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

//classic Spring MVC controller
@RequiredArgsConstructor
@RestController
public class ProductController {
    final ProductRepository repository;

    @PostMapping("/product")
    public Product saveProduct(@RequestBody Product product) {
        return repository.save(product);
    }
}

File: src/main/java/dk/cngroup/fetch/controller/ProductController.kt
package dk.cngroup.fetch.controller

import dk.cngroup.fetch.entity.Product
import dk.cngroup.fetch.entity.ProductRepository
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

//classic Spring MVC controller
@RestController
class ProductController(private val repository: ProductRepository) {

    @PostMapping("/product")
    fun saveProduct(@RequestBody product: Product): Product {
        return repository.save(product)
    }
}